dts/bindings/input/st,stmpe811.yaml

```diff
@@ -0,0 +1,133 @@
+# Copyright (c) 2023 Antmicro <www.antmicro.com>
+# SPDX-License-Identifier: Apache-2.0
+
+description: STMPE811 I2C touchscreen controller
+
+compatible: "st,stmpe811"
+
+include: i2c-device.yaml
+
+properties:
+  int-gpios:
+    type: phandle-array
+    description: |
+      Interrupt GPIO. Used by the controller to signal touch data is
+      available. Active low.
+
+  screen-width:
+    type: int
+    default: 0
+    description: |
+      Screen width for scaling the reported coordinates.
+      Default: raw touchscreen resolution.
+
+  screen-height:
+    type: int
+    default: 0
+    description: |
+      Screen height for scaling the reported coordinates.
+      Default: raw touchscreen resolution.
+
+  raw-x-min:
+    type: int
+    default: 0
+    description: |
+      Signed raw X axis start for scaling the reported coordinates.
+      No effect if screen size is not set.
+      Default: no offset.
+
+  raw-y-min:
+    type: int
+    default: 0
+    description: |
+      Signed raw Y axis start for scaling the reported coordinates.
+      No effect if screen size is not set.
+      Default: no offset.
+
+  raw-x-max:
+    type: int
+    default: 4096
+    description: |
+      Raw X axis end for scaling the reported coordinates.
+      No effect if screen size is not set.
+      Default: raw touchscreen resolution.
+
+  raw-y-max:
+    type: int
+    default: 4096
+    description: |
+      Raw Y axis end for scaling the reported coordinates.
+      No effect if screen size is not set.
+      Default: raw touchscreen resolution.
+
+  panel-driver-settling-time-us:
+    type: int
+    enum:
+      - 10
+      - 100
+      - 500
+      - 1000
+      - 5000
+      - 10000
+      - 50000
+      - 100000
+    required: true
+    description: |
+      Panel driver settling time (microseconds). For large panels (> 6"), a capacitor of 10 nF
+      is recommended at the touchscreen terminals for noise filtering.
+      As a general rule, 1-5 nF capacitors require around 500 us settling time, and 5-10 nF need
+      around 1 ms. When a larger capacitor is used, this value should be changed, as it can
+      lead to inaccuracy of the measurement.
+
+  touch-detect-delay-us:
+    type: int
+    enum:
+      - 10
+      - 50
+      - 100
+      - 500
+      - 1000
+      - 5000
+      - 10000
+      - 50000
+    default: 5000
+    description: |
+      Touch detect delay (microseconds) is the delay from the activation of the pull-up resistor
+      in the X+ line to the time the device performs touch detection.
+      The default value provides enough time for filtering capacitors up to about 50 nF,
+      and above this value, it should be increased.
```

Comment:
not a valid justification. property is board dependent, so it needs to be required.

Reply:
We moved it to the dts.